src/coreclr/vm/threadsuspend.cpp

```diff
@@ -5873,6 +5873,13 @@ void Thread::ApcActivationCallback(ULONG_PTR Parameter)
     ActivationReason reason = (ActivationReason)pData->Parameter;
     PCONTEXT pContext = pData->ContextRecord;
 
+#if defined(TARGET_ARM64)
+    // Windows incorrectly set the CONTEXT_UNWOUND_TO_CALL in the flags of the context it passes to us.
+    // That results in incorrect compensation of PC at some places and sometimes incorrect unwinding 
+    // and GC holes due to that.
+    pContext->ContextFlags &= ~CONTEXT_UNWOUND_TO_CALL;
```

Comment:
Do we need to do the same on NativeAOT?

Reply:
Also why only TARGET_ARM64 - because the issue was only ever seen on ARM64?

Reply:
I don't see that flag used in NativeAOT, so I don't think so.

Reply:
> Also why only TARGET_ARM64 - because the issue was only ever seen on ARM64?

The flag is arm64 only

Reply:
> I don't see that flag used in NativeAOT, so I don't think so.

We pass the context to the platform unwinder. If the flag is incorrectly set, could it possibly confuse the platform unwinder as well?

I can't find any good documentation on the flag's purpose.

Reply:
I do not see any uses of this flag in the platform unwinder. This flag is only used by things like EH region matching and we seem to be able to do fine without it in NativeAOT.